modules/imgcodecs/src/grfmt_gif.cpp

```diff
@@ -392,6 +392,7 @@ bool GifDecoder::lzwDecode() {
             if (code < colorTableSize) {
                 imgCodeStream[idx++] = (uchar)code;
             } else {
+                if (idx + lzwExtraTable[code].length > width * height) return false;
```

Comment:
I would say that it should be just assert like:
```
CV_Assert(idx + lzwExtraTable[code].length <= width * height);
```
The caller side has `try/catch` to handle it. But it's could be caught by debugger.
Some warning about broken gif is useful too.